libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs

```diff
@@ -133,7 +135,24 @@ public async Task ContinueConversationAsync(ConversationReference reference, Bot
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot bot, CancellationToken cancellationToken)
         {
-            if (!_facebookClient.VerifySignature(request))
+            if (request.Query["hub.mode"] == "subscribe")
+            {
+                await _facebookClient.VerifyWebhookAsync(request, response, cancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            await FacebookHelper.WriteAsync(response, HttpStatusCode.OK, string.Empty, Encoding.UTF8, cancellationToken).ConfigureAwait(false);
+
+            string body;
+
+            using (var sr = new StreamReader(request.Body))
+            {
+                body = sr.ReadToEnd();
+            }
+
+            var facebookBody = JsonConvert.SerializeObject(body);
+
+            if (!_facebookClient.VerifySignature(request, body))
```

Comment:
do we need to pass the body here since it is in the request and doesn't seem to be used anywhere else outside this method?

Reply:
In case the answer is yes, a different name such as stringifiedBody or similar would be good not to get people thinking this is directly the request body

Reply:
Yes, because it's a StreamReader and it can only be read once and we'll need it later to process the message. I'll rename it.

Reply:
Done. Thanks!